server/controller/apis/resourceController.js

```diff
@@ -193,13 +193,26 @@ exports.saveResourceImage = async( req, res, resourceId, filename ) => {
 exports.saveResource = async ( req, res, redirect ) => {
 
     let resource = Resource.emptyResource();
+
+    // get the user id either from the request user from basic auth in API call, or from the session for the UI
+    let authUserId;
+    if( req.user ) {
+        authUserId = req.user.id;
+    }
+    else if( req.session.authUser ) {
+        authUserId = req.session.authUser.id;
+    }
+
+    if(authUserId > 0) {
+
     resource.id = req.body.resourceId;
 
     // see if this is a modification of an existing resource
-    let existingResource = await resourceService.getResourceById( resource.id );
+    let existingResource = await resourceService.getResourceById( resource.id, false );
```

Comment:
Without assigning the new resource the ID from req.body.id, I worry that this will never return a matching resource. The previous code had `resource.id = req.body.resourceId` - I think the proper fix would have something like `resource.id = req.body.id` based on model/resource.js

@briangormanly you're far more familiar than we are - do you think this is necessary, or should this work as-is?

Reply:
looking at this

Reply:
I think it is correct because it is pulling from the view
In edit-resource.ejs the line:
<input type="hidden" name="resourceId" id="resourceId" value="<%- resource.id %>" />

Reply:
<input type="hidden" name="resourceId" id="resourceId" value="<%- resource.id %>" />

Reply:
Apparently github does not let you past html.. makes sense I guess.

Reply:
I found the line - makes sense when coming in from the front-end side, but how would this play out if only using the API with Postman/ThunderClient? Our tests for POSTing to the route create new resources when passed an existing resource as JSON. Do you suspect that issue is unrelated to this?